lib/fluent/plugin/output.rb

```diff
@@ -310,6 +310,10 @@ def configure(conf)
             log.warn "'flush_at_shutdown' is false, and buffer plugin '#{buf_type}' is not persistent buffer."
             log.warn "your configuration will lose buffered data at shutdown. please confirm your configuration again."
           end
+
+          if @flush_mode != :interval && buffer_conf.has_key?('flush_interval')
```

Comment:
It looks better to set `:interval` when `flush_mode` is `:default`, even when `timekey` is specified, here:
https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/output.rb#L292
We need to know `flush_interval` is set in `<buffer>` section or not before setting `:interval`... how about to add `has_flush_interval` before `super` as same with `has_buffer_section`?